src/main/java/org/folio/kafka/services/KafkaAdminClientService.java

```diff
@@ -0,0 +1,103 @@
+package org.folio.kafka.services;
+
+import io.vertx.core.Context;
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import io.vertx.kafka.admin.KafkaAdminClient;
+import io.vertx.kafka.admin.NewTopic;
+import org.apache.logging.log4j.Logger;
+import org.folio.kafka.KafkaConfig;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static io.vertx.kafka.admin.KafkaAdminClient.create;
+import static org.apache.logging.log4j.LogManager.getLogger;
+import static org.folio.kafka.services.KafkaEnvironmentProperties.environment;
+import static org.folio.kafka.services.KafkaEnvironmentProperties.replicationFactor;
+
+public class KafkaAdminClientService {
+
+  private static final Logger log = getLogger(KafkaAdminClientService.class);
+  private final Supplier<KafkaAdminClient> clientFactory;
+
+  public KafkaAdminClientService(Vertx vertx) {
+    this.clientFactory = () -> create(vertx, KafkaConfig.builder()
+      .kafkaHost(KafkaEnvironmentProperties.host())
+      .kafkaPort(KafkaEnvironmentProperties.port())
+      .build().getProducerProps());
+  }
+
+  public Future<Void> createKafkaTopics(KafkaTopic[] enumTopics, String tenantId) {
+    final List<NewTopic> topics = readTopics(enumTopics, tenantId)
+      .map(topic -> topic.setReplicationFactor(replicationFactor()))
+      .collect(Collectors.toList());
+    final KafkaAdminClient kafkaAdminClient = clientFactory.get();
+    return createKafkaTopics(1, topics, kafkaAdminClient)
+      .eventually(x -> kafkaAdminClient.close())
+      .onSuccess(result -> log.info("Topics created successfully"))
+      .onFailure(cause -> log.error("Unable to create topics", cause));
+  }
+
+  public Future<Void> deleteKafkaTopics(KafkaTopic[] enumTopics, String tenantId) {
+    List<String> topicsToDelete = readTopics(enumTopics, tenantId)
+      .map(NewTopic::getName)
+      .collect(Collectors.toList());
+    return withKafkaAdminClient(kafkaAdminClient -> kafkaAdminClient.deleteTopics(topicsToDelete))
+      .onSuccess(x -> log.info("Topics deleted successfully"))
+      .onFailure(e -> log.error("Unable to delete topics", e));
+  }
+
+  private Future<Void> createKafkaTopics(int attempt, List<NewTopic> topics, KafkaAdminClient kafkaAdminClient) {
+    return kafkaAdminClient.listTopics()
+      .compose(existingTopics -> {
+        final List<NewTopic> newTopics = new ArrayList<>(topics);
+        newTopics.removeIf(t -> existingTopics.contains(t.getName()));
+        if (newTopics.isEmpty()) {
+          return Future.succeededFuture();
+        }
+        return kafkaAdminClient.createTopics(newTopics);
+      })
+      .recover(e -> {
+        if (!(e instanceof org.apache.kafka.common.errors.TopicExistsException)) {
+          return Future.failedFuture(e);
+        }
+        if (attempt >= 30) {
```

Comment:
should we make it configurable?

Reply:
So far there is no need. I just moved logic from others modules to here:
mod-inventory-storage: [KafkaAdminClientService.java](https://github.com/folio-org/mod-inventory-storage/blob/master/src/main/java/org/folio/services/kafka/topic/KafkaAdminClientService.java)
mod-circulation-storage: [KafkaAdminClientService.java](https://github.com/folio-org/mod-circulation-storage/blob/master/src/main/java/org/folio/service/kafka/topic/KafkaAdminClientService.java)